rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -55,252 +39,37 @@ class Airdrop(NamedTuple):
-        asset=A_CRV,
-        url='https://www.curve.fi/',
-        name='Curve Finance',
-        icon='curve.png',
-    ),
-    'convex': Airdrop(
-        csv_url='https://gist.githubusercontent.com/LefterisJP/fd0ebccbc645f7de2b142907bd207363/raw/0613689dd5212b81788ed1a108c751b29b2ce93a/convex_airdrop.csv',
-        asset=A_CVX,
-        url='https://www.convexfinance.com/',
-        name='Convex',
-        icon='convex.jpeg',
-    ),
-    'shapeshift': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/shapeshift.csv',
-        asset=A_FOX,
-        url='https://shapeshift.com/shapeshift-decentralize-airdrop',
-        name='ShapeShift',
-        icon='shapeshift.svg',
-    ),
-    'ens': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/ens.csv',
-        asset=A_ENS,
-        url='https://claim.ens.domains/',
-        name='ENS',
-        icon='ens.svg',
-    ),
-    'psp': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/psp.csv',
-        asset=A_PSP,
-        url='https://paraswap.io/',
-        name='ParaSwap',
-        icon='paraswap.svg',
-    ),
-    'sdl': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/saddle_finance.csv',
-        asset=A_SDL,
-        url='https://saddle.exchange/#/',
-        name='SaddleFinance',
-        icon='saddle-finance.svg',
-    ),
-    'cow_mainnet': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/cow_mainnet.csv',
-        asset=A_VCOW,
-        url='https://cowswap.exchange/#/claim',
-        name='COW (ethereum)',
-        icon='cow.svg',
-    ),
-    'cow_gnosis': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/cow_gnosis.csv',
-        asset=A_GNOSIS_VCOW,
-        url='https://cowswap.exchange/#/claim',
-        name='COW (gnosis chain)',
-        icon='cow.svg',
-    ),
-    'diva': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/diva.csv',
-        asset=A_DIVA,
-        url='https://claim.diva.community/',
-        name='DIVA',
-        icon='diva.svg',
-    ),
-    'shutter': Airdrop(
-        csv_url='https://raw.githubusercontent.com/rotki/data/main/airdrops/shutter.csv',
-        asset=A_SHU,
-        url='https://claim.shutter.network/',
-        name='SHU',
-        icon='shutter.png',
-        cutoff_time=Timestamp(1721000000),  # Sun Jul 14 2024 23:33:20 UTC
-    ),
-}
+def fetch_airdrops_metadata() -> tuple[dict[str, Airdrop], dict[str, list[str]]]:
+    try:
+        response = requests.get(url=AIRDROPS_INDEX, timeout=10).json()
+        airdrops: dict[str, Airdrop] = {
+            protocol_name: Airdrop(**airdrop_data)
+            for protocol_name, airdrop_data in response['airdrops'].items()
```

Comment:
can't the constructor fail if you provide less than expected/ more fields?

Reply:
I believe the file should include some kind of versioning. What if we change the format of the airdrops for one release? We would break the older releases. Maybe would be better to rename the file to index_v1.json()